clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -167,7 +167,7 @@
  * called <i>test</i> as described above.
  * <p>
  * The broker will automatically detect failed processes in the <i>test</i> group by using a heartbeat mechanism. The
- * consumer will automatically ping the cluster periodically, which lets the cluster know that it is alive. As long as
+ * consumer will ping the cluster whenever {@link #poll(long)} is called, which lets the cluster know that it is alive. As long as
```

Comment:
This is also not quite right. Heartbeats _are_ sent periodically, but they are only triggered in a call to poll(). Maybe we can leave this sentence as it is and add something like the following:

> Note that the consumer is single-threaded, so periodic heartbeats can only be sent when poll() is called. In particular, if the consumer doesn't call poll() before the expiration of the session timeout, then no heartbeats will be sent and the cluster will mark the consumer as failed.


Or something like that. What do you think?